src/validation/timestamp.cairo

```diff
@@ -12,7 +13,27 @@ pub fn validate_timestamp(prev_timestamps: Span<u32>, block_time: u32) -> Result
     // So we cannot assume the list of previous timestamps is sorted.
     // We still want to preserve the "queue" semantics, so we cannot do sorted insert.
     // Thus we need to sort the items in the prev_timestamps before picking a median.
-    if block_time > *prev_timestamps.at(prev_timestamps.len() - 6) {
+    // sort the prev_timestamps
+    let mut sorted_prev_timestamps: Array<u32> = array![];
+    let mut visited: Felt252Dict<bool> = Default::default();
+    while sorted_prev_timestamps.len() != prev_timestamps.len() {
```

Comment:
It seems way too ineffective, I suggest just taking the bubble sort from https://github.com/keep-starknet-strange/alexandria/blob/main/packages/sorting/src/bubble_sort.cairo which will give you a single pass in case of an already sorted array.